### DISCUSSION TITLE
Allow usage of more ram as shared memory for gpu.

### DISCUSSION THREAD

**Discussion by DuckersMcQuack** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#100] [Ideas]
I want to make 16:9/21:9 images, but also for as high base resolution i can, but even with 960x960 static resolution, it cancels and fails because i don't have enough memory. But for regular generations, it can generate just fine with shared memory. But not create tensorRT engines for said resolutions. Please make it able to use shared ram for generating higher res engines.

**Comment by contentis** [comment]
The idea here is to use RAM as swap only to keep parts of the tensors on GPU?

If so, you will not see any performance improvements from TensorRT. PCIe bandwidth and memory in general is the most limiting factor.

**Reply by DuckersMcQuack** [reply]
the idea was kind of that, but all of it in original idea. Because the "training" for the TRT for a specific res can use as long as it needs to, as i saw even with 2048x2048 with latent upscale, it generates fine and fast under memory limit, but as soon as the final VAE decode was it, that's when it will shoot up in memory for 10 sec, but the actual generation uses quite less, which is why i want to use "ram/swap" for training/converting. So it's training for higher res generation profile that needs "past vram limit" removed, or increased.

**Comment by wizz13150** [comment]
Hey, take a look at :

https://nvidia.custhelp.com/app/answers/detail/a_id/5490

**Reply by DuckersMcQuack** [reply]
that's not what i want. I want it to use ram as video memory even if slower. That's the whole idea. But for some reason, it uses more now, but converting has halted when i tried 4096. And it's been 20 min with no change. Could this be fixed somehow? Where like normal generations, it can use shared ram just fine anthough slow, but it's processing. While i convert tensorRT, it's just frozen. 

![image](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/9039370/ac379d13-38cb-4d0d-91d5-d8b528440cba)
